### DISCUSSION TITLE
Error at line 1, column 22

### DISCUSSION THREAD

**Discussion by juamarCas** [nlohmann/json#3160] [Q&A] [answered]
Hi everyone, 
I have made a project based on the rabbitmq broker (AMQP protocol) for transmitting information between devices/processes, so I have a producer that sends information to the broker and the consumer that consumes that information.<br/>
to test this JSON library with this project I decided just to send 2 variables in a JSON format and did the following: 
```C++
nlohmann::json j;
   int var1 = 15; 
   float var2 = 5.5; 

   j["pos"]  = var1; 
   j["temp"] = var2; 

   std::string msg = j.dump(); 
   AMQP::TcpChannel channel(&connection);
   channel.declareQueue("myqueue").onSuccess([&channel, msg](const std::string &name, uint32_t messageCount, uint32_t consumercount){
            std::cout<<"declared queue: myqueue"<<std::endl;
            while(1){        
               std::cin.get();  
               channel.publish("", "myqueue", msg); 
            }
   });
```
Everything here works fine but is just to give context. In my Consumer.cpp file I have the following script to test that the information arrives without any problem: 

```C++
auto messageCb = [&channel](const AMQP::Message &message, uint64_t deliveryTag, bool redelivered) {

    std::cout << "message received:" << std::endl;
    std::cout.write(message.body(), message.bodySize())<<std::endl;  
    channel.ack(deliveryTag);  

    char j_receive[message.bodySize()];
    strncpy(j_receive, message.body(), message.bodySize()); 

    std::string j_final = std::string(j_receive);
    auto j_parsed = nlohmann::json::parse(j_final); 
    std::cout<<j_parsed.dump(4)<<j_receive<<"\n\n"; 
   
};
```
What I did is just copy the message that arrived from the broker and paste it in a char * variable, and then made it as a string. this is the output from the cout: 
```
message received:
{"pos":15,"temp":5.5}
```
and then, when I use the parse() function to the string i have the following exception: 
```
erminate called after throwing an instance of 'nlohmann::detail::parse_error'
  what():  [json.exception.parse_error.101] parse error at line 1, column 22: syntax error while parsing value - invalid literal; last read: '5.5}'; expected end of input
Aborted (core dumped)
```

I don't know if there is a problem with the process I use to the message that is arriving.

Thanks a lot.

**Comment by nlohmann** [comment, marked as answer]
On first sight, this could be a problem with a missing null byte in your `char*`. Can you double check the string is properly null-terminated, and maybe execute your code with ASAN enabled to check if all bounds are properly kept?

**Reply by juamarCas** [reply]
Hi, Sorry for late reply.
I tried appending a '\0' at the end of the string but the error still pops. I will try later with ASAN what information can give me.

Thanks

**Reply by juamarCas** [reply]
Ok, I did the following: 

```C++
    char j_receive[message.bodySize() + 1];
    strncpy(j_receive, message.body(), message.bodySize() + 1); 
    j_receive[message.bodySize()] = '\0'; 
```

and worked perfectly, it was the null terminated.

Thanks a lot
